frontend/src/pages/add-blog.tsx

```diff
@@ -111,24 +111,31 @@ function AddBlog() {
     }
   };
   const navigate = useNavigate();
-
-  const [isDarkMode, setIsDarkMode] = useState(
-    window.matchMedia('(prefers-color-scheme: dark)').matches
-  );
-
+  const [isDarkMode, setIsDarkMode] = useState<boolean | null>(null)
+    //  window.matchMedia('(prefers-color-scheme: dark)').matches
+  ;
   useEffect(() => {
-    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+    // Check if the user has set a theme preference in local storage
+    const storedTheme = localStorage.getItem('theme');
+    //const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
 
-    const handleThemeChange = (event: MediaQueryListEvent) => {
-      setIsDarkMode(event.matches);
-    };
+    // Set initial theme based on local storage or system preference
+    setIsDarkMode(storedTheme === 'dark');
+  }, []);
+  // useEffect(() => {
+  //   const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
 
-    mediaQuery.addListener(handleThemeChange);
+  //   const handleThemeChange = (event: MediaQueryListEvent) => {
+  //     setIsDarkMode(event.matches);
+  //   };
 
-    return () => {
-      mediaQuery.removeListener(handleThemeChange);
-    };
-  }, []);
+  //   mediaQuery.addListener(handleThemeChange);
+
+  //   return () => {
+  //     mediaQuery.removeListener(handleThemeChange);
+  //   };
+  // }, []);
+  // if(storedTheme === 'dark') setIsDarkMode(true);
 
```

Comment:
I see there are lots of commented code, kindly delete them, and keep only the required comments, tbh comments are actually not required, in a first look seems like you have directly copy pasted from chat-gpt, i may be wrong, but for the first honest opinion it looks that way :)